### PR TITLE
Fix data fetching

### PR DESCRIPTION
**Pull request by adrianstanescu**

### CODE REVIEW

File: src/backend/riot/api.ts [hunk 1]
```diff
@@ -1,24 +1,17 @@
...
     const cached = cache.get(CacheSection.APISummoner, key);
     if (cached) {
         return new Summoner(account, cached as Partial<Summoner>);
     }
-    const response = await riotFetch(
-        `${baseURL(account.server)}/summoner/v4/summoners/by-name/${
-            account.name
-        }`
-    );
+    const response = await riotFetch(APIRequestMethod.SummonerByName, account.name);
```
Comment (tehciolo) [inline src/backend/riot/api.ts:14]: Ah! Much cleaner!

File: src/backend/riot/client.ts [hunk 2]
```diff
@@ -1,16 +1,55 @@
...
+                return this.fetch(method, ...args);
+            } else {
+                throw new Error(`${response.status} ${response.statusText} response from Riot API`);
+            }
+        }
+        return response;
+    }
+}
+
+const DEFAULT_CLIENT = new Client(riotAPIKey());
```
Comment (tehciolo) [inline src/backend/riot/client.ts]: [nitpick] 

I don't consider this a CONSTANT, but rather a `Class` instance.

```suggestion
const defaultClient = new Client(riotAPIKey());
```

File: src/backend/riot/rateLimiter.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        this.appLimits = limits
+            .filter(({ type }) => type === RateLimitType.App)
+            .map(({ seconds, limit }) => new RateLimit(limit, seconds));
+        const methodLimitConfigs = limits.filter(({ type }) => type === RateLimitType.Method);
+        this.methodLimits = Object.fromEntries(
+            Object.keys(APIRequestMethod).map((method) => {
+                return [
+                    method,
+                    methodLimitConfigs.map(
+                        ({ type, seconds, limit }) => new RateLimit(limit, seconds)
```
Comment (tehciolo) [inline src/backend/riot/rateLimiter.ts]: `type` is not used here?

File: src/backend/riot/rateLimiter.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    refreshCounts(method: APIRequestMethod, headers: Headers) {
+        const status = getRateLimitStatus(headers);
+        if (this.hasInitialLimits) {
+            this.setLimits(status);
+            this.hasInitialLimits = false;
+        }
+        for (const info of status) {
+            const limit =
+                info.type === RateLimitType.App
+                    ? this.appLimits.find((l) => l.seconds === info.seconds)
```
Comment (tehciolo) [inline src/backend/riot/rateLimiter.ts]: [nitpick]

I assume `l` stands for limit here, but I prefer having it spelled out.

File: src/backend/riot/rateLimiter.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+                    : this.methodLimits[method]?.find((l) => l.seconds === info.seconds);
+            if (!limit) {
+                continue;
+            }
+            limit.refreshCount(info.count);
+        }
+        if (headers.has('Retry-After')) {
+            const retryAfterMS = parseInt(headers.get('Retry-After') ?? '5', 10) * 1000;
```
Comment (tehciolo) [inline src/backend/riot/rateLimiter.ts]: [nitpick]

I prefer to use the `Number` constructor whenever I need to parse to a base 10 integer. It looks a bit more clean to me.

**Review by tehciolo — CHANGES_REQUESTED** [review]

File: src/backend/riot/rateLimiter.ts (hunk 4 above)
Comment (adrianstanescu) [reply src/backend/riot/rateLimiter.ts]: Fixed this instance and created #6 to address the rest of the code.

**Review by tehciolo — APPROVED** [review]

**Pull request merged**